crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -75,11 +78,26 @@ pub async fn gossip_process_with_start_seq<A>(
     let mut gossip_tasks = FuturesUnordered::new();
 
     loop {
+        if _active_authority.state.committee.load().epoch != committee.epoch {
+            // If epoch has changed, we need to make a new copy of the active authority,
+            // and update all local variables.
+            // We also need to remove any authority that's no longer a valid validator
+            // from the list of peer names.
+            // It's ok to keep the existing gossip tasks running even for peers that are no longer
```

Comment:
Why not do `gossip_tasks.clear()` to kill all the awaiting gossip tasks from the past epoch?

Reply:
Why do we want to do that?